api/mod.go

```diff
@@ -354,9 +354,19 @@ func (g *Gateway) sendTransaction(ctx *fasthttp.RequestCtx) {
 
 	if !tx.VerifySignature() {
 		g.renderError(ctx, ErrBadRequest(errors.New("bad signature")))
+
+		return
+	}
+
+	snapshot := g.ledger.Snapshot()
+
+	if err := wavelet.ValidateTransaction(snapshot, tx); err != nil {
```

Comment:
why we validate tx here instead passing `validateState=true` to `Add` method?

Reply:
I feel that we should always validate the state of transactions within `(*Ledger).AddTransaction`. Rather than pass an optional boolean flag `validateState`, we should enforce it by default.